contracts/contracts/governance/MainnetGovernanceExecutor.sol

```diff
@@ -0,0 +1,279 @@
+pragma solidity ^0.8.0;
+
+import { Governable } from "./Governable.sol";
+import { QUEUE_PROPOSAL_COMMAND, CANCEL_PROPOSAL_COMMAND } from "./L2Governance.sol";
+import { Initializable } from "../utils/Initializable.sol";
+
+import { ARBITRUM_ONE_SELECTOR } from "../utils/CCIPChainSelectors.sol";
+
+import { Client } from "@chainlink/contracts-ccip/src/v0.8/ccip/libraries/Client.sol";
+import { IRouterClient } from "@chainlink/contracts-ccip/src/v0.8/ccip/interfaces/IRouterClient.sol";
+
+contract MainnetGovernanceExecutor is Governable, Initializable {
+    /***************************************
+                    Events
+    ****************************************/
+    /**
+     * @dev Emitted whenever a command is forwarded to CCIP Router
+     */
+    event CommandSentToCCIPRouter(
+        uint64 indexed chainSelector,
+        bytes32 messageId,
+        bytes2 commandSelector,
+        uint256 proposalId
+    );
+    /**
+     * @dev Emitted when a Chain Config is added
+     */
+    event ChainConfigAdded(
+        uint64 indexed chainSelector,
+        address indexed l2Governance
+    );
+    /**
+     * @dev Emitted when a Chain Config is removed
+     */
+    event ChainConfigRemoved(uint64 indexed chainSelector);
+
+    /***************************************
+                    Errors
+    ****************************************/
+    error UnsupportedChain(uint64 chainSelector);
+    error InsufficientBalanceForFees(uint256 feesRequired);
+    error DuplicateChainConfig(uint64 chainSelector);
+    error InvalidGovernanceCommand(bytes2 command);
+    error InvalidInitializationArgLength();
+    error InvalidGovernanceAddress();
+
+    /***************************************
+                    Storage
+    ****************************************/
+    address public immutable ccipRouter;
+
+    struct ChainConfig {
+        bool isSupported;
+        address l2Governance;
+    }
+    /**
+     * @dev All supported chains
+     */
+    mapping(uint64 => ChainConfig) public chainConfig;
+
+    constructor(address _ccipRouter) {
+        ccipRouter = _ccipRouter;
+    }
+
+    function initialize(
+        uint64[] calldata chainSelectors,
+        address[] calldata l2Governances
+    ) public initializer {
+        if (chainSelectors.length != l2Governances.length) {
+            revert InvalidInitializationArgLength();
+        }
+
+        for (uint256 i = 0; i < chainSelectors.length; ++i) {
+            _addChainConfig(chainSelectors[i], l2Governances[i]);
+        }
+    }
+
+    /***************************************
+                    CCIP
+    ****************************************/
+    /**
+     * @dev Send a command to queue/cancel a L2 Proposal through CCIP Router
+     * @param commandSelector Command to send
+     * @param chainSelector Destination chain
+     * @param proposalId L2 Proposal ID
+     * @param maxGasLimit Max Gas Limit to use
+     */
+    function _sendCommandToL2(
```

Comment:
Is this necessary since you also have `queueL2Proposal` and `cancelL2Proposal` and for now `L2Governance` only accepts these two message types?

Reply:
I assume you mean the `sendCommandToL2` and not this internal function? I can get rid of that, yeah

Reply:
misread

Reply:
Have removed that `sendCommandToL2`. Also, removed the `InvalidGovernanceCommand` check and error since those are not user-input now